test/Source/DoctrineORMQuerySourceIteratorTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Exporter\Test\Source;
+
+use Doctrine\ORM\Configuration;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Query;
+use Exporter\Test\Source\Fixtures\DoctrineORMQuerySourceIterator;
+use Exporter\Test\Source\Fixtures\ObjectWithToString;
+
+/**
+ * @author Joseph Maarek <josephmaarek@gmail.com>
+ */
+final class DoctrineORMQuerySourceIteratorTest extends \PHPUnit_Framework_TestCase
+{
+    /** @var Query */
+    protected $query;
+
+    protected function setUp()
+    {
+        /** @var EntityManagerInterface|\PHPUnit_Framework_MockObject_MockObject $entityManager */
+        $entityManager = $this->createMock('Doctrine\ORM\EntityManagerInterface');
+        $entityManager->method('getConfiguration')->willReturn(new Configuration());
+        $this->query = new Query($entityManager);
+    }
+
+    /**
+     * @dataProvider getValueProvider
+     */
+    public function testGetValue($value, $expected, $dateFormat = 'r')
```

Comment:
Add type hint whenever possible

Reply:
Here, it is not possible.
$value and $expected can be of any type
$dateFormat can be string or null, but default is 'r'